sensing/autoware_cuda_pointcloud_preprocessor/include/autoware/cuda_downsample_filter/thrust_custom_allocator.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright 2025 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTOWARE__CUDA_DOWNSAMPLE_FILTER__THRUST_CUSTOM_ALLOCATOR_HPP_
+#define AUTOWARE__CUDA_DOWNSAMPLE_FILTER__THRUST_CUSTOM_ALLOCATOR_HPP_
+
+#include <cuda_runtime.h>
+#include <thrust/device_malloc_allocator.h>
+
+#include <sstream>
+
+namespace autoware::cuda_downsample_filter
+{
+struct ThrustCustomAllocator : public thrust::device_malloc_allocator<uint8_t>
+{
+  // ref: https://stackoverflow.com/questions/76594790/memory-pool-in-thrust-execution-policy
+public:
+  using Base = thrust::device_malloc_allocator<uint8_t>;
+  using pointer = typename Base::pointer;
+  using size_type = typename Base::size_type;
+
+  explicit ThrustCustomAllocator(cudaStream_t stream) : stream_(stream) {}
+
+  pointer allocate(size_type num)
+  {
+    uint8_t * buffer(nullptr);
+    auto result = cudaMallocAsync(&buffer, num, stream_);
```

Comment:
What about https://github.com/autowarefoundation/autoware_universe/blob/main/sensing/autoware_cuda_utils/include/autoware/cuda_utils/cuda_check_error.hpp
?

Reply:
Thanks. In 3ccc84d61, I changed the code to use macro defined in autoware_cuda_utils for CUDA related error check